actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/service/impl/UserServiceImpl.java

```diff
@@ -95,8 +95,14 @@ public User getUserById(String userId) {
 
   @Override
   public void validateUserId(Request request, String managedById) {
+    String userId = null;
     String ctxtUserId = (String) request.getContext().get(JsonKey.USER_ID);
-    String userId = userExtIdentityDao.getUserId(request);
+    if (StringUtils.isEmpty(ctxtUserId)) {
+      ctxtUserId = (String) request.getContext().get(JsonKey.REQUESTED_BY);
+    }
+    if (!request.getOperation().equals(ActorOperations.CREATE_USER_V4.getValue())) {
+      userId = userExtIdentityDao.getUserId(request);
```

Comment:
Why does validateUserId need to go usr external entity table at all in the first place?
If id, provider, idType are null, which I think will be the case getUserId will return empty. What will happen then?